internal/services/storage/storage_account_resource.go

```diff
@@ -1318,6 +1319,153 @@ func resourceStorageAccountCreate(d *pluginsdk.ResourceData, meta interface{}) e
 
 	supportLevel := resolveStorageAccountServiceSupportLevel(storage.Kind(accountKind), storage.SkuTier(accountTier))
 
+	// Wait for the account services to become available
+
+	if supportLevel.supportBlob {
+		blobClient := meta.(*clients.Client).Storage.BlobServicesClient
+
+		// wait for blob service endpoint to become available
+		log.Printf("[DEBUG] waiting for %s blob service to become available", id.ID())
+		deadline, ok := ctx.Deadline()
+		if !ok {
+			return fmt.Errorf("internal-error: context had no deadline")
+		}
+		stateConf := &pluginsdk.StateChangeConf{
+			Pending: []string{"Pending"},
+			Target:  []string{"Completed"},
+			Refresh: func() (interface{}, string, error) {
+				_, err = blobClient.GetServiceProperties(ctx, id.ResourceGroupName, id.StorageAccountName)
+				if err != nil {
+					return handleStorageServiceError("blob", err)
+				}
+				return true, "Completed", nil
+			},
+			MinTimeout: 10 * time.Second,
+			Timeout:    time.Until(deadline),
+		}
+
+		if _, err = stateConf.WaitForStateContext(ctx); err != nil {
+			return fmt.Errorf("fetching %s blob service properties: %+v", id.ID(), err)
+		}
+	}
+
+	if supportLevel.supportQueue {
+		storageClient := meta.(*clients.Client).Storage
+		account, err := storageClient.FindAccount(ctx, id.StorageAccountName)
+		if err != nil {
+			return fmt.Errorf("retrieving %s: %+v", id, err)
+		}
+		if account == nil {
+			return fmt.Errorf("unable to locate %q", id)
+		}
+
+		queueClient, err := storageClient.QueuesClient(ctx, *account)
+		if err != nil {
+			return fmt.Errorf("building Queues Client: %s", err)
+		}
+
+		// wait for queue service endpoint to become available
+		log.Printf("[DEBUG] waiting for %s queue service to become available", id.ID())
+		deadline, ok := ctx.Deadline()
+		if !ok {
+			return fmt.Errorf("internal-error: context had no deadline")
+		}
+		stateConf := &pluginsdk.StateChangeConf{
+			Pending: []string{"Pending"},
+			Target:  []string{"Completed"},
+			Refresh: func() (interface{}, string, error) {
+				properties, err := queueClient.GetServiceProperties(ctx, id.ResourceGroupName, id.StorageAccountName)
+				if err != nil {
+					return handleStorageServiceError("queue", err)
+				}
+				// workaround for queueClient.GetServiceProperties() not returning 404 error
+				if properties == nil {
+					customErr := azautorest.DetailedError{
+						StatusCode: http.StatusNotFound,
+						Message:    "Failure responding to request",
+					}
+					return handleStorageServiceError("queue", customErr)
```

Comment:
this error message is pretty misleading, if there's a DNS replication delay (for example, when recreating a storage account) we'll be indicating that the storage account was available when it wasn't - instead we could simplify this to:

```suggestion
					return false, "Pending", nil
```